syntax = "proto3";
package ant_proto;

import  "hardware.proto";
import  "utils.proto";

message Color {
    uint32 r = 1;
    uint32 g = 2;
    uint32 b = 3;
}

message EntityData {
    int64 x = 1;
    int64 y = 2;

    int32 ch = 3;
    int64 fov_radius = 4;
    Color color = 5;
}

message Building {
    int64 id = 1;
    Rect rect = 2;
    Color col = 3;
}

message Player {
    EntityData data = 1;
    Inventory inventory = 2;
}

message Worker {
    EntityData data = 1;
    Inventory inventory = 2;
    ProgramExecutor program_executor = 3;
    DualRegisters dual_registers = 4;
    
}

message EntityManager {
    Player player = 1;
    uint64 player_depth = 6;
}

message Inventory {
    uint64 max_stack_count = 1;
    uint64 stack_size = 2;
    uint64 max_weight = 3;
    // maps item type to item count
    repeated InventoryRecord item_records = 4;
}

message InventoryRecord {
    // enum ItemType {
    //     DIRT=0;
    //     FOOD=1;
    //     EGG=2;
    // }
    uint64 type = 1;
    uint64 count = 2;
}

